SAM parser

I tried reading a sam file:

```julia
julia> reader = open(SAMReader, "1to2k.sam")
ERROR: AssertionError: upanchor! called with no anchor set
 in upanchor! at /Users/bdeonovic/.julia/v0.5/Bio/src/Ragel.jl:33 [inlined]
 in macro expansion at /Users/bdeonovic/.julia/v0.5/Bio/src/Ragel.jl:104 [inlined]
 in macro expansion at /Users/bdeonovic/.julia/v0.5/Bio/src/align/hts/sam/parser.jl:1299 [inlined]
 in _read!(::Bio.Align.SAMReader, ::Bio.Ragel.State{BufferedStreams.BufferedInputStream{IOStream}}, ::Bio.Align.SAMRecord) at /Users/bdeonovic/.julia/v0.5/Bio/src/Ragel.jl:182
 in read(::Bio.Align.SAMReader) at /Users/bdeonovic/.julia/v0.5/Bio/src/Ragel.jl:228
 in Bio.Align.SAMReader(::BufferedStreams.BufferedInputStream{IOStream}) at /Users/bdeonovic/.julia/v0.5/Bio/src/align/hts/sam/reader.jl:20
 in #open#2(::Array{Any,1}, ::Function, ::Type{Bio.Align.SAMReader}, ::String) at /Users/bdeonovic/.julia/v0.5/Bio/src/IO.jl:58
 in open(::Type{Bio.Align.SAMReader}, ::String) at /Users/bdeonovic/.julia/v0.5/Bio/src/IO.jl:58
```

However, after converting the SAM to BAM and using the BAMReader I had no problem reading the file.

--- a/src/align/hts/sam/reader.jl
+++ b/src/align/hts/sam/reader.jl
@@ -17,6 +17,7 @@ type SAMReader <: Bio.IO.AbstractReader
         reader = new(Ragel.State(samparser_start, input), SAMHeader())
         while !eof(input) && BufferedStreams.peek(input) == UInt8('@')
             # NOTE: This reads a header line, not the first record.
+            reader.state.cs = samparser_start
             @assert read(reader) == SAMRecord()
         end
         return reader

